crates/sui-network/src/randomness/mod.rs

```diff
@@ -101,6 +104,43 @@ impl Handle {
             .expect("RandomnessEventLoop mailbox should not overflow or be closed")
     }
 
+    /// Admin interface handler: generates partial signatures for the given round at the
+    /// current epoch.
+    pub fn admin_get_partial_signatures(
+        &self,
+        round: RandomnessRound,
+        tx: oneshot::Sender<Vec<u8>>,
+    ) {
+        self.sender
+            .try_send(RandomnessMessage::AdminGetPartialSignatures(round, tx))
+            .expect("RandomnessEventLoop mailbox should not overflow or be closed")
+    }
+
+    /// Admin interface handler: injects partial signatures for the given round at the
+    /// current epoch, skipping validity checks.
+    pub fn admin_inject_partial_signatures(
+        &self,
+        authority_name: AuthorityName,
+        round: RandomnessRound,
+        sigs: Vec<RandomnessPartialSignature>,
+    ) {
+        self.sender
+            .try_send(RandomnessMessage::AdminInjectPartialSignatures(
+                authority_name,
+                round,
+                sigs,
+            ))
+            .expect("RandomnessEventLoop mailbox should not overflow or be closed")
+    }
+
+    /// Admin interface handler: injects full signature for the given round at the
+    /// current epoch, skipping validity checks.
+    pub fn admin_inject_full_signature(&self, round: RandomnessRound, sig: RandomnessSignature) {
+        self.sender
+            .try_send(RandomnessMessage::AdminInjectFullSignature(round, sig))
```

Comment:
might be best if this (and the previous message) took a `tx` so they can send back a Result. This way the admin route can send the error (if any) back to the user.

Reply:
Why can't full sig error? couldn't the sig be bad?

Reply:
It can't error inside the randomness network loop bc we already have it deserialized into the fastcrypto type here